client/src/util/helpers.ts

```diff
@@ -305,4 +306,26 @@ export default {
       return translation + " " + translationSuffix;
     }
   },
+
+  /**
+   * Detects whether a given DOM element is visible on screen.
+   * Borrowed from https://stackoverflow.com/questions/45514676/react-check-if-element-is-visible-in-dom
+   */
+  useOnScreen(ref: React.RefObject<any>) {
+    const [isIntersecting, setIntersecting] = useState(false);
+
+    const observer = new IntersectionObserver(([entry]) => setIntersecting(entry.isIntersecting));
```

Comment:
Hmm, I'm noticing that the current portfolio tab seems to work (mostly) on IE11 but this PR makes the portfolio tab crash on this line, because IE11 doesn't support `IntersectionObsever`.  In general, [older browsers from 2018 don't support it](https://caniuse.com/intersectionobserver) so it might be worth using a progressive enhancement strategy here, e.g. checking to see if `typeof IntersectionObserver === "undefined"` and returning some kind of default value if it's unsupported.

Reply:
Good catch! Will add this in now, and adjust the description of the function to match.